File: lint-arwaky/src-python/taxonomy/git_diff_vo.py
"""git_diff_vo — VO for git diff summary."""

from pydantic import BaseModel, ConfigDict
from .path_collection_vo import FilePathList, RenamedFileList
from .message_status_vo import Count


class GitDiffResultVO(BaseModel):
    """Aggregated diff result for surface display."""

    model_config = ConfigDict(frozen=True)

    added: FilePathList
    modified: FilePathList
    deleted: FilePathList
    renamed: RenamedFileList
    lintable_files: FilePathList
    all_files: FilePathList
    total_changed: Count

File: lint-arwaky/src-rust/taxonomy/git_diff_vo.rs
use serde::{Serialize, Deserialize};
use std::collections::{HashMap, HashSet};
use super::*;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct GitDiffResultVO {
    pub added: FilePathList,
    pub modified: FilePathList,
    pub deleted: FilePathList,
    pub renamed: RenamedFileList,
    pub lintable_files: FilePathList,
    pub all_files: FilePathList,
    pub total_changed: Count,
}

impl GitDiffResultVO {
    pub fn new(added: FilePathList, modified: FilePathList, deleted: FilePathList, renamed: RenamedFileList, lintable_files: FilePathList, all_files: FilePathList, total_changed: Count,) -> Self {
        Self { added, modified, deleted, renamed, lintable_files, all_files, total_changed }
    }
}
